noun/trace.c

```diff
@@ -295,6 +298,180 @@ u3t_flee(void)
   u3z(don);
 }
 
+static FILE* trace_file = NULL;
+static int nock_pid = 0;
+
+/*  u3t_trace_open(): opens a trace file and writes the preamble.
+*/
+void
+u3t_trace_open(c3_c* trace_file_name)
+{
+  printf("trace: tracing to %s\n", trace_file_name);
+  trace_file = fopen(trace_file_name, "w");
+  nock_pid = (int)getpid();
+  fprintf(trace_file, "[ ");
```

Comment:
It doesn't look like we ever close this`[`. Is that ok?

Reply:
For that matter, there doesn't seem to be a `u3t_trace_close()`, which I think should exist and get called on graceful shutdown (Ctrl-D).

Reply:
I've added a `u3t_trace_close()` function which closes the file. It doesn't write a closing `]` because of the file format.